packages/js/src/utils/readApiConnection.ts

```diff
@@ -0,0 +1,268 @@
+// TODO(jon): Pretty sure this whole file should just be a separate package that gets packaged
+// alongside the Read API instead
+
+import { TokenStandard } from '@metaplex-foundation/mpl-token-metadata';
+import { PROGRAM_ID as BUBBLEGUM_PROGRAM_ID } from '@metaplex-foundation/mpl-bubblegum';
+import {
+  Commitment,
+  Connection,
+  ConnectionConfig,
+  PublicKey,
+} from '@solana/web3.js';
+import BN from 'bn.js';
+import type { Metadata, Mint, NftOriginalEdition } from '@/plugins';
+import type { SplTokenCurrency } from '@/types';
+import { Pda, amount, toBigNumber } from '@/types';
+
+import { ReadApiError } from '@/errors/ReadApiError';
+import type {
+  GetAssetProofRpcInput,
+  GetAssetProofRpcResponse,
+  GetAssetRpcInput,
+  GetAssetsByOwnerRpcInput,
+  GetAssetsByGroupRpcInput,
+  ReadApiAsset,
+  ReadApiAssetList,
+} from '@/types/ReadApi';
+
+type JsonRpcParams<ReadApiMethodParams> = {
+  method: string;
+  id?: string;
+  params: ReadApiMethodParams;
+};
+
+type JsonRpcOutput<ReadApiJsonOutput> = {
+  result: ReadApiJsonOutput;
+};
+
+export const toNftEditionFromReadApiAsset = (
+  input: ReadApiAsset
+): NftOriginalEdition => {
+  return {
+    model: 'nftEdition',
+    isOriginal: true,
+    address: new PublicKey(input.id),
+    supply: toBigNumber(input.supply.print_current_supply),
+    maxSupply: toBigNumber(input.supply.print_max_supply),
+  };
+};
+
+export const toMintFromReadApiAsset = (input: ReadApiAsset): Mint => {
+  const currency: SplTokenCurrency = {
+    symbol: 'Token',
+    decimals: 0,
+    namespace: 'spl-token',
+  };
+
+  return {
+    model: 'mint',
+    address: new PublicKey(input.id),
+    // TODO(jon): Presumably, this should be the Master Edition address upon decompression
```

Comment:
@lorisleiva just want to confirm this. Read API does not provide this but we can derive it, if this is indeed supposed to be the Master Edition address that would show up upon decompression

Reply:
I think this is good enough for now. I'm a bit worried about setting this to the edition PDA since this would only make sense for non-fungible token standards.